src/main/java/frc/trigon/robot/commands/AlignToNoteCommand.java

```diff
@@ -0,0 +1,87 @@
+import frc.trigon.robot.misc.objectdetectioncamera.ObjectDetectionCamera;
+import frc.trigon.robot.subsystems.ledstrip.LEDStripCommands;
+import frc.trigon.robot.subsystems.ledstrip.LEDStripConstants;
+import frc.trigon.robot.subsystems.swerve.SwerveCommands;
+import org.trigon.utilities.mirrorable.MirrorableRotation2d;
+
+import java.awt.*;
+
+public class AlignToNoteCommand extends ParallelCommandGroup {
+    private static final ObjectDetectionCamera CAMERA = CameraConstants.NOTE_DETECTION_CAMERA;
+    private static final PIDController Y_PID_CONTROLLER = new PIDController(0.005, 0, 0);
+    private boolean didCollect = false;
+    private boolean wasVisible = false;
+    private double trackedNoteYaw = 0;
+
+    public AlignToNoteCommand() {
+        configureNoteCollectionDetectionTrigger();
+        addCommands(
+                new InstantCommand(() -> {
+                    didCollect = false;
+                    wasVisible = false;
+                }),
+                getCurrentLEDColorCommand().asProxy(),
+                Commands.getContinuousConditionalCommand(getDriveWhileAligningToNoteCommand(), Commands.duplicate(CommandConstants.SELF_RELATIVE_DRIVE_COMMAND), this::hasTarget).asProxy(),
+                new RunCommand(this::trackObject)
+        );
+    }
+
+    private void trackObject() {
+        if (hasTarget() && !wasVisible) {
+            wasVisible = true;
+            CAMERA.startTrackingBestObject();
+            trackedNoteYaw = CAMERA.getTrackedObjectYaw();
+            return;
+        }
+
+        if (!hasTarget()) {
+            wasVisible = false;
+            return;
+        }
+
+        if (hasTarget())
+            trackedNoteYaw = CAMERA.getTrackedObjectYaw();
+    }
+
+    private Command getCurrentLEDColorCommand() {
+        return Commands.getContinuousConditionalCommand(
+                LEDStripCommands.getStaticColorCommand(Color.green, LEDStripConstants.LED_STRIPS),
+                LEDStripCommands.getStaticColorCommand(Color.red, LEDStripConstants.LED_STRIPS),
+                CAMERA::hasTargets
+        );
+    }
+
+    private Command getDriveWhileAligningToNoteCommand() {
+        return SwerveCommands.getClosedLoopSelfRelativeDriveCommand(
+                () -> CommandConstants.calculateDriveStickAxisValue(OperatorConstants.DRIVER_CONTROLLER.getLeftY()),
+                () -> Y_PID_CONTROLLER.calculate(-trackedNoteYaw),
+                this::getTargetAngle
+        );
+    }
+
+    private MirrorableRotation2d getTargetAngle() {
+        final Rotation2d currentRotation = RobotContainer.POSE_ESTIMATOR.getCurrentPose().getRotation();
+        return new MirrorableRotation2d(currentRotation.plus(Rotation2d.fromDegrees(trackedNoteYaw)), false);
+    }
+
+    private void configureNoteCollectionDetectionTrigger() {
+        if (RobotContainer.INTAKE.isEarlyNoteCollectionDetected())
+            new InstantCommand(() -> didCollect = true);
+    }
```

Comment:
this has multiple problems
1. this will happen when the function is called, not when the command happens
2. you're never scheduling the command 
3. the name suggests this is a trigger